### ISSUE TITLE
Schematic overview

### ISSUE THREAD

**Issue by FireflySE** [larry-athey/rpi-smart-still#2]
![Skärmbild 2024-03-04 233629](https://github.com/larry-athey/rpi-smart-still/assets/98463572/ac454377-b40c-4fe9-ae18-6f96cd5d0ebf)

Sorry for all the mess I tried to follow your instructions and made a schematic for the hole system except the scr and heating element.
If you have the time please look it over and give me input.

**Comment by FireflySE** [comment]
[Uploading RPI Smart Still.zip…]()

Puting the fritzing file here as well

**Comment by larry-athey** [comment]
Thanks! You obviously have way more patience for that Fritzing program than I have. I think it's because their parts library doesn't contain all the parts that I use. Does Fritzing have a way to use the schematic in order export a Gerber file that could be used to make printed circuit boards?

**Comment by FireflySE** [comment]
Yes its not that good you have to improvise..  I think EasyEDA or Autodesk Eagle is better, but for this project and because the breadboard overview its easy to follow. I wanted to contribute with something to the project at least. I haven´t used Fritzing that much either so to export to a gerber file a guess that´s possible but I would use a better software for that. My plan for the future is to make a gerber and order a nice pcb but for now I want to test and play with your fine project.

**Comment by larry-athey** [comment]
Everything looks correct to me. I'd like to put these in the /Diagrams/ folder here and create a README.md to describe what's what. How do you want me to credit you for these? FireflySE or some other name?

Also, I just pushed an update so you will want to sync your code again. I discovered an oversight in the process.php that saves the DS18B20 sensor addresses.

**Comment by FireflySE** [comment]
FireflySE is fine thank you. And will sync the code 👍

**Comment by FireflySE** [comment]
oh as you see on the schematic i used 12v system if you want to change it to 24v please do.

**Comment by FireflySE** [comment]
Updated schematic missed power and ground to one of the logic level shifters

**Comment by FireflySE** [comment]
Hi, couldn´t help it I made a prototype pcb with some extra features it needs to be verified and reworked but this is what i came up with 
![Skärmbild 2024-03-09 132055](https://github.com/larry-athey/rpi-smart-still/assets/98463572/72eabe4e-26b8-4591-a6b5-786772d1c8f7)

**Comment by FireflySE** [comment]
![image](https://github.com/larry-athey/rpi-smart-still/assets/98463572/8708a040-331e-4afb-9468-f163dee3f478)

**Comment by larry-athey** [comment]
_What are the extra features, and what is the long J1 connector for?_

Edit: Ahhh, it's a parallel connection of the GPIO bus. What are the two relays on GPIO 17/27 for?

**Comment by FireflySE** [comment]
The 2 relays is in my case for controlling 2 12v water pumps. There are 2 3pin screw connectors on the board for this.
Then i added led fault and power indicators.

**Comment by FireflySE** [comment]
![Skärmbild 2024-03-10 151846](https://github.com/larry-athey/rpi-smart-still/assets/98463572/a88e39c1-b4ce-48ef-bd65-92677e5b0693)
Updated layout, smaller board 100x90 and 4 layers.

**Comment by FireflySE** [comment]
Updated with more common components and footprints
![Skärmbild 2024-03-12 003535](https://github.com/larry-athey/rpi-smart-still/assets/98463572/58e6546d-a9d7-43a3-9372-3dd329f6d8e7)

**Comment by FireflySE** [comment]
New update with more common components and footprints. I will order the first prototype and let you know how it goes.
![Skärmbild 2024-03-12 010356](https://github.com/larry-athey/rpi-smart-still/assets/98463572/69bef389-5ec2-416c-9afb-90e285024ec7)

**Comment by larry-athey** [comment]
What's your plan for connecting the temperature sensors?

**Comment by FireflySE** [comment]
Good point they need 5v and ground according to your schematic I missed this and should be 4.7k resistors on the data line I have corrected this now and made a tmp module with connector for data lines +5v and ground. Thanks for the observation

**Comment by FireflySE** [comment]
![Skärmbild 2024-03-12 191040](https://github.com/larry-athey/rpi-smart-still/assets/98463572/df003bda-57e2-41df-aa9b-c77cf5cbc1cf)

**Comment by FireflySE** [comment]
![Skärmbild 2024-03-12 191134](https://github.com/larry-athey/rpi-smart-still/assets/98463572/a558497a-b0ee-4aa4-8c31-ea7c60ed9556)

**Comment by FireflySE** [comment]
![Skärmbild 2024-03-12 195218](https://github.com/larry-athey/rpi-smart-still/assets/98463572/276f15f0-3368-45b0-81fd-c57a78342d70)

**Comment by larry-athey** [comment]
Oh, I just figured you had some other kind of external device planned to connect them and the serial lines to J1. Each temperature sensor doesn't need a pull up resistor, there just needs to be one between +5 and GPIO 4. All of the sensors are connected in parallel to the GPIO pin since the PI identifies them by their address.

**Comment by FireflySE** [comment]
Aaah ok yes that make sense will update the schematic.

**Comment by FireflySE** [comment]
![Screenshot 2024-03-13 093931](https://github.com/larry-athey/rpi-smart-still/assets/98463572/d3281625-28f3-4900-b0b6-a149d56ddd72)

**Comment by FireflySE** [comment]
![Screenshot 2024-03-13 094113](https://github.com/larry-athey/rpi-smart-still/assets/98463572/cd584be3-8ac0-4d68-9cfa-4ab5e7202eae)

**Comment by FireflySE** [comment]
![Skärmbild 2024-03-13 190535](https://github.com/larry-athey/rpi-smart-still/assets/98463572/5c835b07-047c-4c9b-af17-4d54d0f9f203)

**Comment by larry-athey** [comment]
OK, that works. You could actually eliminate one of those DS18B20 connections because there are only sensors on the boiler, column, and dephlegmator discharge water. The one that's on the digital hydrometer is connected directly to its own ESP32. All communications from the digital hydrometer to the PI is done over the serial tx/rx GPIO pins 14 and 15. I'd call it "a protocol" but it's just a matter of parsing formatted text and sending a couple single byte values to the hydrometer to reboot it or recalibrate it.

**Comment by FireflySE** [comment]
Nice thanks I added an extra connection only for if you want an extra sensor somewhere but maybe its unnecessary..

**Comment by FireflySE** [comment]
I think I am done tinkering with the board now and ready to order a prototype. Will let you know how much magic smoke I got ;) 

![Skärmbild 2024-03-15 171723](https://github.com/larry-athey/rpi-smart-still/assets/98463572/4746a55f-a6b5-45b3-8c60-f2d4b643be97)

**Comment by larry-athey** [comment]
Wow, I'm guessing that this was all done in EasyEDA. I have yet to give that a try, I just haven't had the time to mess with it. I can see that you've driven completely around the use of the modules that I use and duplicated them right on the PCB. That's probably another reason why I haven't tried working with it, I'm still not sure of the lifespan of them so I'm still hung up on using things that can be replaced at the modular level.

Let me know if it keeps all of the magic smoke inside! LOL!!!

**Comment by FireflySE** [comment]
Yes I used EasyEda for this, because of low use hours if you destill on hobby basis I hope it will last a long time. But I will check temperature of components and make necessary adjustments if need be. The cost of the board at jlcpcb is 7$ for me to get it to Sweden about 20$. If it works as intended without too much magic smoke I will upload the gerber file. 🤣

**Comment by larry-athey** [comment]
Just checking in to see what the status is on the PCB. I also wanted to ask if I'm overlooking the serial connector for the hydrometer data communications. I'm in the middle of building a simplified alternative that uses LIDAR to read the floating height of a glass hydrometer in a normal parrot. It will still have the temperature sensor on the input, but the flow sensor will be moved to the parrot output.

**Comment by FireflySE** [comment]
Hi, 
Interesting project similar to what George D. did if I recall correctly. I have been way to busy with work unfortunately. The test pcb have arrived I assembled the pcb but it shorts out some where "no smoke"😉 have not yet been been able to test it further. I only noticed that the drv8825 ic got warm. I can send you the gerber file if you want to take a look. What do you mean by overlooking serial connection?
I cant see what could cause it to not work with current setup🤔

br Mathias

**Comment by larry-athey** [comment]
I was referring to the two UART pins 14 and 15 on the GPIO bus, that's how the PI and the hydrometer talk to each other. I didn't see screw terminals for them, or I overlooked them and can't figure out which ones they are.

My board with the DRV8825 has a decent size heat sink on it and it's always a bit warm even when it's sitting idle.

George's "talking parrot head" uses the same type of sensor, but that PVC tube on top of the parrot is entirely too bulky. You can do the same thing with the sensor attached to the side of the parrot aiming upward and not make it top heavy. Mine also has the flow sensor on the output of the parrot and the temperature sensor on the input side. This will be easier to build and calibrate, I spooked too many people with the load cell based digital hydrometer.

**Comment by FireflySE** [comment]
Ok yes I have not added a separate connector for 14-15 and you're right that would be nice to have on the board. I can add this to the next test pcb. They are on the rpi output pins on my schematic. 

I agree your solution is better 👍, But I fell in love with your loadcell solution so I think i will use that but nice to have options.

I work with EBF 3D printing and we are in the works with building a new industrial machine and we work alot even weekends so sorry for not being fast to respond. If your interested take a look at www.freemelt.com.

Br Mathias

**Comment by larry-athey** [comment]
I wish there were companies like that around here, I'd love to have a job in that type of industry.

**Comment by FireflySE** [comment]
Iam sorry to hear that.. We actually have a branch on US soil now and we hope to expand with the new "america makes" investment. 

Keep in touch 

Br Mathias

**Comment by larry-athey** [comment]
Those relays you added to GPIO 17 and 27...Should I send those pins high at the start of a run and then low at the end? Or are you implementing something else to control them?

**Comment by FireflySE** [comment]
My thought for those were pumps and yes one of them could be high when temp is above maybe 40-45C and above in a normal run. Nr 2 is for reflux and should be high when reflux program is selected. If you dont have controllable valves you can just run 2 pumps was my idea. Could be done with one pump and a manual shut off valve for the reflux. But would be nice to control temp in both condenser and reflux column. If you dont want to add this to your code then its ok 😊

**Comment by larry-athey** [comment]
This system would be completely non-functional without the valves and limit switches, the Start Run procedure wouldn't even start if you haven't run the valve calibration routine. Speaking of which, if those relays are for pumps, they would need to be active even if a run isn't active because you need to calibrate the valves under water pressure I've tested the calibration with and without water pressure and the numbers are definitely different.

**Comment by FireflySE** [comment]
Yes then it would be best as you say to have them on high at the start of the run and low at the end👍

**Comment by larry-athey** [comment]
I'll add a page under the Management menu to toggle them on an off. I can use at least one of them for controlling the power for my water pressure regulator.

**Comment by FireflySE** [comment]
Sweet! Is there anything else you want to be added to the board?

**Comment by larry-athey** [comment]
No, the serial communications terminals were the only thing that I could see that was needed. I'll be anxious to rebuild my controller with the hat in a smaller form factor.

**Comment by FireflySE** [comment]
Ok I will try to figure out the issues soon and send you a functional gerber.Skickat från min Galaxy
-------- Originalmeddelande --------Från: Larry Athey ***@***.***> Datum: 2024-05-11  23:10  (GMT+01:00) Till: larry-athey/rpi-smart-still ***@***.***> Kopia: FireflySE ***@***.***>, Author ***@***.***> Ämne: Re: [larry-athey/rpi-smart-still] Schematic overview (Issue #2) 
No, the serial communications terminals were the only thing that I could see that was needed. I'll be anxious to rebuild my controller with the hat in a smaller form factor.

**Comment by larry-athey** [comment]
OK, the "Control Relays" page is in there and the last on/off state of each relay is restored across reboots.

At this point, I think I'm ready for an official first official GitHub release number of the source code. I intended to have this done by December 31, 2023 but I got my ass kicked by a blizzard that messed up all kinds of my plans. I ran through test distillations of all four of the programs I've included and there wasn't a single issue with any of them.
